### PR TITLE
ci: enable basic linters in golangci-lint

### PR DESCRIPTION
**Pull request by Madhu-1**
This PR enables a few basic linters in golangci-lint.

fixes https://github.com/csi-addons/volume-replication-operator/issues/128
Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

### CODE REVIEW

File: .github/workflows/config/lint-config.yml [hunk 1]
```diff
@@ -108,7 +108,7 @@ linters-settings:
     # max line length, lines longer will be reported. Default is 120.
     # '\t' is counted as 1 character by default, and can be changed with the
     # tab-width option
-    line-length: 120
+    line-length: 180
```
Comment (iamniting) [inline .github/workflows/config/lint-config.yml:111]: I think this change should be part of the first commit itself, where these changes were introduced.

**Review by iamniting — CHANGES_REQUESTED** [review]

File: .github/workflows/config/lint-config.yml (hunk 1 above)
Comment (Madhu-1) [reply .github/workflows/config/lint-config.yml]: done.

**Comment by iamniting** [comment]
@Madhu-1 The PR looks good to me, Was CI complaining to run tests parallely?

**Comment by Madhu-1** [comment]
> @Madhu-1 The PR looks good to me, Was CI complaining to run tests parallely?

Yes, linter complains about it that's the reason for running tests parallelly.

**Review by iamniting — DISMISSED** [review]

File: .github/workflows/codespell.yaml [hunk 2]
```diff
@@ -16,7 +16,7 @@ jobs:
       - name: codespell
         uses: codespell-project/actions-codespell@master
         with:
-          skip: .git
+          skip: .git,vendor
```
Comment (sp98) [inline .github/workflows/codespell.yaml:19]: also skip `go.mod` and `go.sum` ?

Comment (Madhu-1) [reply .github/workflows/codespell.yaml:19]: sometimes we add comments in go.mod i can skip go.sum as its auto generated

**Review by iamniting — APPROVED** [review]

**Review by sp98 — APPROVED** [review]

**Comment by Madhu-1** [comment]
@mergifyio refresh

**Comment by Madhu-1** [comment]
@Mergifyio refresh

**Pull request merged**